conformance/tests/httproute-invalid-reference-grant.go

```diff
@@ -98,13 +98,13 @@ var HTTPRouteInvalidReferenceGrant = suite.ConformanceTest{
 		// at least one allowed BackendRef should be accepted by a Gateway
 		// and partially configured.
 
-		t.Run("Simple HTTP request should not reach app-backend-v2", func(t *testing.T) {
+		t.Run("HTTP Request to invalid backend with missing referenceGrant should receive a 500", func(t *testing.T) {
```

Comment:
Returning a 500 here as distinct from the not found/unmatched 404 behavior resolves the TODO on line 90 - `routeNN` should be added as an additional argument to `kubernetes.GatewayAndHTTPRoutesMustBeReady` because the route should be `Accepted` for this difference to happen.